internal/core/model.rs

```diff
@@ -954,6 +954,38 @@ impl<C: RepeatedComponent> Repeater<C> {
         crate::item_tree::VisitChildrenResult::CONTINUE
     }
 
+    /// Move to the next focus item in the repeater
+    pub fn next_in_focus_chain(&self, subindex: Option<u32>, result: &mut crate::items::ItemWeak) {
```

Comment:
What is subindex?  What does it mean to be None ?

Also you should return the result. Using a `&mut` for the result is not idiomatic rust at all. 
It is only used for `extern "C"` function that cannot return types with destructor in C++

Reply:
I need a way to differentiate between the different components below the repeater. So I just pass the `index` that the repeater had assigned to each of those tries and use that. I am not particularly happy with the name...

`index` refers to indices into the item tree everywhere else in this context, so I did not want to go with that. I went with `subindex`, as it is the `index` of the repeater pointed to by the `index`.

`None` is basically I don't know whether there are any sub-trees yet.

Good point about the `result`, I need to change that. I started with the same signature everywhere (with the C++ portability quirks) and already fixed some to be nicer to use.